client/src/components/RelatedItems/Related.jsx

```diff
@@ -1,55 +1,78 @@
-import React, { useState } from 'react';
-import { FaArrowAltCircleRight, FaArrowAltCircleLeft } from 'react-icons/fa';
+import React, { useState, useEffect, useContext } from 'react';
+import { FaArrowAltCircleRight, FaArrowAltCircleLeft, FaStar } from 'react-icons/fa';
+import { CombinedAPIDetails } from './RelatedItemsHooks';
 
-function Related({ relatedItems, handleClick }) {
-  const [arr, setArr] = useState([0, 1, 2, 3]);
-  const length = relatedItems.length;
+function Related({ setProductId }) {
+  const combinedAPIDetails = useContext(CombinedAPIDetails);
 
-  const nextSlide = () => {
-    setArr(
-      arr[3] !== length - 1
-        ? [arr[0] + 1, arr[0] + 2, arr[0] + 3, arr[0] + 4]
-        : [length - 4, length - 3, length - 2, length - 1]
+  const referenceArray = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+  useEffect(() => {
+    console.log("componentDidMout time????")
+    setShownImagesArray([0, 1, 2, 3]);
+    setShownImagesOffset(0)
+  }, [combinedAPIDetails]);
+
+  const [shownImagesArray, setShownImagesArray] = useState([]);
+  const [shownImagesOffset, setShownImagesOffset] = useState(0);
```

Comment:
Should these state declarations be moved up above the useEffect? 
Wouldn't it cause an error to have them declared _after_ they're used in the useEffect block?